datasafe-rest-impl/src/main/java/de/adorsys/datasafe/rest/impl/controller/UserController.java

```diff
@@ -41,7 +43,10 @@ public void createUser(@RequestBody UserDTO requestDTO) {
         ReadKeyPassword readKeyPassword = new ReadKeyPassword(requestDTO.getPassword());
         UserIDAuth auth = new UserIDAuth(new UserID(requestDTO.getUserName()), readKeyPassword);
 
-        dataSafeService.userProfile().registerUsingDefaults(auth);
+        if (dataSafeService.userProfile().userExists(auth.getUserID())) {
+            throw new RestImplException("user \"" + auth.getUserID().getValue() + "\" already exists");
```

Comment:
Makes sense to throw more specific exception like UserExistsException

Reply:
ok.